enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_count
+    return length unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = 0
+    i = 0
+    while i < arr.size
+      count += 1 if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = []
+    i = 0
+    while i < arr.size
+      count << arr[i] if yield(arr[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_all?
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      return false unless yield(arr[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?(con = false)
+    if block_given?
+      size.times do |item|
+        result = yield(self[item])
+        (return true) if result
+      end
+    elsif con
+      size.times do |item|
+        return true if con == self[item]
+      end
+    else
+      size.times do |item|
+        return true if self[item]
+      end
+    end
+    false
+  end
+
+  def my_none?
+    unless block_given?
+      my_each { |item| return false if item != false }
+      return true
+    end
+    my_each do |item|
+      return false if yield(item) == true
+    end
+    true
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[false, nil, false].my_none? == [false, nil, false].none? => true
```
- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/79677159-63160680-81ab-11ea-8324-fb3cd752b2e6.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/79677164-71642280-81ab-11ea-8366-8db04c80be89.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677170-9193e180-81ab-11ea-855b-d261347166b1.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677183-bdaf6280-81ab-11ea-8ea9-cce8d0e0b2dd.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677191-d3bd2300-81ab-11ea-9efd-cd6e21111e14.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:
```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?

Reply:
my_none method has been fixed!